openpype/pipeline/colorspace.py

```diff
@@ -0,0 +1,464 @@
+import re
+import os
+import sys
+import json
+import platform
+import contextlib
+import tempfile
+from openpype import PACKAGE_DIR
+from openpype.settings import get_project_settings
+from openpype.lib import (
+    StringTemplate,
+    run_openpype_process,
+    Logger
+)
+from openpype.pipeline import Anatomy
+
+log = Logger.get_logger(__name__)
+
+
+@contextlib.contextmanager
+def _make_temp_json_file():
+    """Wrapping function for json temp file
+    """
+    try:
+        # Store dumped json to temporary file
+        temporary_json_file = tempfile.NamedTemporaryFile(
+            mode="w", suffix=".json", delete=False
+        )
+        temporary_json_file.close()
+        temporary_json_filepath = temporary_json_file.name.replace(
+            "\\", "/"
+        )
+
+        yield temporary_json_filepath
+
+    except IOError as _error:
+        raise IOError(
+            "Not able to create temp json file: {}".format(
+                _error
+            )
+        )
+
+    finally:
+        # Remove the temporary json
+        os.remove(temporary_json_filepath)
+
+
+def get_ocio_config_script_path():
+    """Get path to ocio wrapper script
+
+    Returns:
+        str: path string
+    """
+    return os.path.normpath(
+        os.path.join(
+            PACKAGE_DIR,
+            "scripts",
+            "ocio_wrapper.py"
+        )
+    )
+
+
+def get_imageio_colorspace_from_filepath(
+    path, host_name, project_name,
+    config_data=None, file_rules=None,
+    project_settings=None,
+    validate=True
+):
+    """Get colorspace name from filepath
+
+    ImageIO Settings file rules are tested for matching rule.
+
+    Args:
+        path (str): path string, file rule pattern is tested on it
+        host_name (str): host name
+        project_name (str): project name
+        config_data (dict, optional): config path and template in dict.
+                                      Defaults to None.
+        file_rules (dict, optional): file rule data from settings.
+                                     Defaults to None.
+        project_settings (dict, optional): project settings. Defaults to None.
+        validate (bool, optional): should resulting colorspace be validated
+                                   with config file? Defaults to True.
+
+    Returns:
+        str: name of colorspace
+    """
+    if not any([config_data, file_rules]):
+        project_settings = project_settings or get_project_settings(
+            project_name
+        )
+        config_data = get_imageio_config(
+            project_name, host_name, project_settings)
+        file_rules = get_imageio_file_rules(
+            project_name, host_name, project_settings)
+
+    # match file rule from path
+    colorspace_name = None
+    for _frule_name, file_rule in file_rules.items():
+        pattern = file_rule["pattern"]
+        extension = file_rule["ext"]
+        ext_match = re.match(
+            r".*(?=.{})".format(extension), path
+        )
```

Comment:
Should the extension potentially need to go through `re.escape()`?

Reply:
Yeah, that could be also possible, but since I am using raw string this should be solid.